doc/source/dev/contributor/debugging_linalg_issues.rst

```diff
@@ -0,0 +1,440 @@
+challenging mathematically/algorithmically (that is true for many parts of
+SciPy), but because BLAS/LAPACK libraries are a complex build-time as well as
+runtime dependency - and we support a significant number of BLAS/LAPACK
+libraries.
+
+This document aims to provide guidance about how to go about debugging linear
+algebra issues. 
+
+If there is a real bug, it can be in one of three places:
+
+- In the BLAS library being used,
+- In SciPy's bindings for BLAS or LAPACK (generated by ``numpy.f2py`` and/or
+  Cython),
+- In SciPy's algorithmic code.
+
+A key first step is to determine whether the bug is in SciPy or in the BLAS
+library. To do so, the most efficient way to disambiguate the two is to set up
+your environment in such a way that you can achieve runtime switching between
+different BLAS libraries (something we don't support out of the box, and isn't
+possible with SciPy's wheels from PyPI).
+
+Upstream BLAS library authors strongly prefer to get clean reproducers (just
+like we do), which means: no Python involved. So this guide will also cover how
+to create reproducers in C or Fortran.
+
+
+Finding the BLAS library being used
+-----------------------------------
+
+SciPy has one function, `~scipy.show_config`, to introspect the build
+configuration of an installed package. This allows querying for details of BLAS
+and LAPACK. E.g.::
+
+    >>> blas_dep = scipy.show_config(mode='dicts')['Build Dependencies']['blas']
+    >>> for key in blas_dep:
+    ...     print(f"{key}:  {blas_dep[key]}")
+    ...
+    name:  openblas
+    found:  True
+    version:  0.3.23
+    detection method:  pkgconfig
+    include directory:  /home/user/mambaforge/envs/scipy-dev/include
+    lib directory:  /home/user/mambaforge/envs/scipy-dev/lib
+    openblas configuration:  USE_64BITINT=0 DYNAMIC_ARCH=1 DYNAMIC_OLDER= NO_CBLAS= NO_LAPACK=0 NO_LAPACKE= NO_AFFINITY=1 USE_OPENMP=0 PRESCOTT MAX_THREADS=128
+    pc file directory:  /home/user/mambaforge/envs/scipy-dev/lib/pkgconfig
+
+This method will be correct for SciPy wheels and for local dev builds. It *may*
+be correct for other installs, however keep in mind that distros like
+conda-forge and Debian may be building against stub libraries (typically
+``blas.so``/``lapack.so``) and then installing another library for the user -
+in such cases, plain ``blas`` and ``lapack`` will be reported even if for
+example OpenBLAS or MKL is installed in the environment. For such installs,
+`threadpoolctl <https://github.com/joblib/threadpoolctl>`__ will usually be
+able to report what the actual BLAS library in use is::
+
+    $ python -m threadpoolctl -i scipy.linalg
+    [
+      {
+        "user_api": "blas",
+        "internal_api": "openblas",
+        "prefix": "libopenblas",
+        "filepath": "/home/user/mambaforge/envs/dev/lib/libopenblasp-r0.3.21.so",
+        "version": "0.3.21",
+        "threading_layer": "pthreads",
+        "architecture": "SkylakeX",
+        "num_threads": 24
+      }
+    ]
```

Comment:
Trying on netlib lapack from conda-forge, the naming is a bit unusual:

```
$ python -m threadpoolctl -i scipy.linalg
[
  {
    "user_api": "openmp",
    "internal_api": "openmp",
    "num_threads": 8,
    "prefix": "libgomp",
    "filepath": "/home/br/mambaforge/envs/scipy-dev-netlib/lib/libgomp.so.1.0.0",
    "version": null
  }
]
```

ldd reports:

```
$ ldd build/scipy/linalg/_flapack.cpython-311-x86_64-linux-gnu.so
	linux-vdso.so.1 (0x00007ffe7b7c1000)
	liblapack.so.3 => /home/br/mambaforge/envs/scipy-dev-netlib/lib/liblapack.so.3 (0x00007f330e7c7000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f330e5bf000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f330f0d9000)
	libblas.so.3 => /home/br/mambaforge/envs/scipy-dev-netlib/lib/./libblas.so.3 (0x00007f330e54c000)
	libgfortran.so.5 => /home/br/mambaforge/envs/scipy-dev-netlib/lib/./libgfortran.so.5 (0x00007f330e3a1000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f330e252000)
	libgomp.so.1 => /home/br/mambaforge/envs/scipy-dev-netlib/lib/./libgomp.so.1 (0x00007f330e214000)
	libgcc_s.so.1 => /home/br/mambaforge/envs/scipy-dev-netlib/lib/./libgcc_s.so.1 (0x00007f330e1f9000)
	libquadmath.so.0 => /home/br/mambaforge/envs/scipy-dev-netlib/lib/./libquadmath.so.0 (0x00007f330e1c0000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f330e19d000)
	librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007f330e193000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f330e18b000)
```

Reply:
> Trying on netlib lapack from conda-forge, the naming is a bit unusual:

That's not it. The command reports also on `openmp` libraries. And in this case fails to report on Netlib BLAS: https://github.com/joblib/threadpoolctl/issues/159.

Reply:
Added a note linking to that issue.